### ISSUE TITLE
Autopilot overuses fuel during long-range cruising

### ISSUE THREAD

**Issue by caligari87** [pioneerspacesim/pioneer#4191]
Not sure if this has been reported yet. I searched under "fuel" and "autopilot" and didn't find anything that matched my experience, but I may have missed it.

### Observed behaviour
Per the title, the autopilot uses too much fuel for micro-corrections during long-range cruising (outside acceleration or deceleration). On short hops (less than ~4-5AU) this is entirely negligible. On trips from outside the system (such as after hyperspace jumps) where the distance is longer (more than ~10-15AU), this is extremely dangerous because there seems to be no safe limiter and the autopilot will happily microburn all the fuel during the week or so it takes to fly into the system, leaving none available for deceleration or maneuvering.

Generally what I have to do is "fly to vicinity of target", timelapse until the ship is up to speed, then disable the autopilot and drift until I'm within 2-3 AU before reactivating it. Otherwise I run out of fuel or crash.

### Expected behaviour
It seems these 00.01% fuel burns are attempting to compensate for the relative orbital movement of the target, since it can take several days or weeks to fly into the system from a hyperspace jump. I believe the autopilot should ignore "small" orbital corrections at excessive ranges, or something. In any case it should *not* simply burn fuel doing corrections until the ship is much closer to the target.

### Steps to reproduce
* Get to some starting point far outside a system (more than ~10-15AU). Hyperspace jumps usually leave me at this range.
* Tell the autopilot to fly to some target in the system.
* Increase timescale to max.
* Watch your fuel slowly decrease over the next several days ingame.
* On longer trips or with target orbits that oscillate a lot relative to your flight path, you'll very likely drop below the available delta-v to stop in time, and be left drifting or crash. 

__My pioneer version (and OS):__
20171001, Linux 64-bit (Ubuntu/Mint)

**Comment by impaktor** [comment]
agreed with @nozmajner. Although, we do need current exposed "perfect" autopilot to be (close as possible to) "perfect" since it's the same that AI uses. But functionallity exposed to player should be lower in accuracy.

Maybe @jaj22 has insights?

**Comment by fluffyfreak** [comment]
Disagree with these views, mostly because although the goal is laudable it misses out on the AI and that the player should have the option and not be hobbled by bad code ;)

Ignoring deviations below a threshold sounds like the quickest and easiest way of helping with this issue. A small planet in a fast orbit at a long distance away shouldn't cause course corrections until you get much closer and have slowed down. 
For example we could apply this filter/test to only happen when a ship is within the root frame and greater than some arbitrary distance from it's target.

**Comment by caligari87** [comment]
I can see the logic behind wanting to encourage more engagement for the player, but I disagree that the player should be *forced* to do it. Personally I find long-range cruising kinda stressful (maybe because I'm still bad at it, but anyway). Planning, trading, shopping, mission selection, and landing/takeoffs are where I have the most fun. I'd rather set the autopilot with confidence and make a sandwich while it cruises to vicinity, then handle the small stuff myself.

Would there be anything wrong with making it an option? Have 2-3 autopilots available for purchase, and each one has better fuel efficiency and/or route-planning. Easy start gets a better one, hard start gets the worst one or none. 

* Dumb Autopilot: Points at target and accelerates/decelerates as needed. Not very fuel efficient, but fine for in-system planet hopping. Can't auto-dock or land.  Targets a 1/3 fuel reserve but doesn't account for small corrections.
* Enhanced Autopilot: More fuel-efficient, avoids burning fuel during long-range cruising into or out of system. Able to auto-dock and land. Tries to maintain a 1/4 fuel reserve, accounting for corrections while getting you there quicker.
* Next-Gen Autopilot: Able to plot semi-complex maneuvers such as gravitational slingshots if they'd be faster or more fuel efficient. Only aims for a 1/10th fuel reserve, so it can get you anywhere quicker than the others. Requires zero pilot intervention after takeoff and course setting.

**Comment by impaktor** [comment]
I think we're drifting from the topic. 

What should be discussed here is the observed errors of the current auto pilot, and since it is the same as the AI it shouldn't have any discernable error.

**Comment by Another-MrZ** [comment]
A separate but related issue is that when the fuel runs out, no warning appears on screen, and the time acceleration does not return automatically to 1x. If you have to manually refuel your ship, you want to do it before you drift too much away from your destination.

**Comment by lolo101** [comment]
Updated link to learn how to [fly manually](https://wiki.pioneerspacesim.net/wiki/How_to_fly_without_autopilot) :wink:

**Comment by DamagedEngine** [comment]
> A separate but related issue is that when the fuel runs out, no warning appears on screen, and the time acceleration does not return automatically to 1x. If you have to manually refuel your ship, you want to do it before you drift too much away from your destination.

That also happens in this related issue (#5552 ) which occurs when the autopilot is re-targeted to a new location which is slightly off the current course.
